Src/IronPython/Runtime/LiteralParser.cs

```diff
@@ -48,9 +48,14 @@ public static class LiteralParser {
                         int len = (ch == 'u') ? 4 : 8;
                         int max = 16;
                         if (isUniEscape) {
-                            if (TryParseInt(text, i, len, max, out val)) {
+                            if (TryParseInt(text, i, len, max, out val, out int consumed)) {
                                 if (val < 0 || val > 0x10ffff) {
-                                    throw PythonExceptions.CreateThrowable(PythonExceptions.UnicodeDecodeError, isRaw ? "rawunicodeescape" : "unicodeescape", Bytes.Empty, i - start - 2, i - start + len - 1, "illegal Unicode character");
+                                    throw PythonExceptions.CreateThrowable(
+                                        PythonExceptions.UnicodeDecodeError,
+                                        isRaw ? "rawunicodeescape" : "unicodeescape",
+                                        PythonOps.MakeBytes(new String(text)),
```

Comment:
Thanks for your opinion. This line is exactly the reason why this PR is not ready yet for merging — I am not happy with how the bytes are reconstructed here but need more time to think about options and implications.

Reply:
The fix is probably have a byte[] version of the method and use that in `UnicodeEscapeEncoding` instead.

Reply:
That would eliminate the superfluous conversion to `char[]`, now done just for the sake of calling `LiteralParser.ParseString`. It would be nice it `ParseString` (or at least the core of it) could be made generic, rather than duplicating the whole thing. 

How is fan level towards `PythonOps.MakeString(IList<byte> bytes)`?

Reply:
Not a fan, but it doesn't seem as bad since it doesn't throw...